pkg/api/v1alpha1/cloudstackmachineconfig_types.go

```diff
@@ -117,6 +117,51 @@ func (c *CloudStackMachineConfig) OSFamily() OSFamily {
 	return ""
 }
 
+func (c *CloudStackMachineConfigSpec) Equal(o *CloudStackMachineConfigSpec) bool {
+	if c == o {
+		return true
+	}
+	if c == nil || o == nil {
+		return false
+	}
+	if !c.Template.Equal(&o.Template) ||
+		!c.ComputeOffering.Equal(&o.ComputeOffering) {
+		return false
+	}
+	if c.Affinity != o.Affinity {
+		return false
+	}
+	if len(c.AffinityGroupIds) != len(o.AffinityGroupIds) ||
+		len(c.Users) != len(o.Users) ||
+		len(c.UserCustomDetails) != len(o.UserCustomDetails) {
+		return false
+	}
+	for i, groupId := range c.AffinityGroupIds {
+		if groupId != o.AffinityGroupIds[i] {
+			return false
+		}
+	}
+	for i, user := range c.Users {
+		if user.Name != o.Users[i].Name {
+			return false
+		}
+		if len(user.SshAuthorizedKeys) != len(o.Users[i].SshAuthorizedKeys) {
+			return false
+		}
+		for j, key := range user.SshAuthorizedKeys {
+			if key != o.Users[i].SshAuthorizedKeys[j] {
```

Comment:
Same with checking for slices here as above:

https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L450

Reply:
I found UsersSliceEqual and will use that instead